### PR TITLE
Add gallery example for grdlandmask

### PR DESCRIPTION
**Pull request by michaelgrund**
**Description of proposed changes**

According to #1421 here's now the gallery example showcasing the use of ``grdlandmask``.

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: examples/gallery/images/grdlandmask.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+region = [-65, -40, -40, -20]
+
+# Assign a value of 0 for all water masses and a value of 1 for all land masses
+grid = pygmt.grdlandmask(region=region, spacing="5m", maskvalues=[0, 1], resolution="l")
+
+# Plot clipped grid
+fig.basemap(region=region, projection="M12c", frame=True)
+
+fig.grdimage(grid=grid, cmap="lajolla")
+fig.colorbar(position="JMR+o0.5c/0c+w8c")
```
Comment (seisman) [inline examples/gallery/images/grdlandmask.py:23]: I think it makes more sense to have a discrete CPT file, with 0 and 1 representing water and land, respectively.

You can refer to the `makecpt` call at this [example](https://www.pygmt.org/v0.4.1/gallery/symbols/points_categorical.html).

File: examples/gallery/images/grdlandmask.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+"""
+
+import pygmt
+
+fig = pygmt.Figure()
+
+# Define region of interest
+region = [-65, -40, -40, -20]
+
+# Assign a value of 0 for all water masses and a value of 1 for all land masses
```
Comment (seisman) [inline examples/gallery/images/grdlandmask.py:16]: Also need to explain the `spacing` and `resolution` parameters.

File: examples/gallery/images/grdlandmask.py (hunk 1 above)
Comment (michaelgrund) [reply examples/gallery/images/grdlandmask.py]: Done!

File: examples/gallery/images/grdlandmask.py (hunk 2 above)
Comment (michaelgrund) [reply examples/gallery/images/grdlandmask.py]: Done!

File: examples/gallery/images/grdlandmask.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import pygmt
+
+fig = pygmt.Figure()
+
+# Define region of interest
+region = [-65, -40, -40, -20]
+
+# Assign a value of 0 for all water masses and a value of 1 for all land masses.
+# Use shoreline data with (l)ow resolution and set the grid spacing to 5 m in x
+# and y direction.
```
Comment (seisman) [inline examples/gallery/images/grdlandmask.py:18]:
```suggestion
# Use shoreline data with (l)ow resolution and set the grid spacing to 5 arc minute
# in x and y direction.
```

**Review by seisman — APPROVED** [review]

**Pull request merged**